cocos/core/math/vec3.ts

```diff
@@ -355,6 +356,77 @@ export class Vec3 extends ValueType {
         return out;
     }
 
+    /**
+     * @zh 球面线性插值。多用于插值两个方向向量。
+     * @en Spherical linear interpolation. Commonly used in interpolation between directional vectors.
+     * @param out @zh 出口向量。 @en Output vector.
+     * @param from @zh 起点向量。 @en Start vector.
+     * @param to @zh 终点向量。 @en Destination vector.
+     * @param t @zh 插值参数。@en Interpolation parameter.
+     * @returns `out`
+     * @description
+     * @zh
+     * - 如果 `from`、`to` 中任何一个接近零向量，则结果就是 `from` 到 `to` 线性插值的结果；
+     *
+     * - 否则，如果 `from`、`to` 方向刚好接近相反，
+     * 则结果向量是满足以下条件的一个向量：结果向量和两个输入向量的夹角之比是 `t`，其长度是 `from` 到 `to` 的长度线性插值的结果；
+     *
+     * - 否则，结果是从标准化后的 `from` 到 标准化后的 `to`
+     * 进行球面线性插值的结果乘以 `from` 到 `to` 的长度线性插值后的长度。
+     * @en
+     * - If either `from` or `to` is close to zero vector,
+     * the result would be the (non-spherical) linear interpolation result from `from` to `to`.
+     *
+     * - Otherwise, if `from` and `to` have almost opposite directions,
+     * the result would be such a vector so that:
+     * The angle ratio between result vector and input vectors is `t`,
+     * the length of result vector is the linear interpolation of lengths from `from` to `to`.
+     *
+     * - Otherwise, the result would be the spherical linear interpolation result
+     * from normalized `from` to normalized `to`,
+     * then scaled by linear interpolation of lengths from `from` to `to`.
+     */
+    public static slerp= (() => {
```

Comment:
Do we need to account for spherical interpolation of non-unit vectors?

Reply:
This can implement, for example:

![image](https://user-images.githubusercontent.com/28807867/188540557-1562713e-a284-4534-8a02-a086e7ef7ec6.png)